src/streaming/models/CmcdModel.js

```diff
@@ -191,28 +202,149 @@ function CmcdModel() {
         }
     }
 
+    function isCmcdEnabled() {
+        const cmcdParametersFromManifest = getCmcdParametersFromManifest();
+        return _canBeEnabled(cmcdParametersFromManifest) && _checkIncludeInRequests(cmcdParametersFromManifest) && _checkAvailableKeys(cmcdParametersFromManifest);
+    }
+
+    function _canBeEnabled(cmcdParametersFromManifest) {
+        if (Object.keys(cmcdParametersFromManifest).length) {
+            if (!cmcdParametersFromManifest.version) {
+                logger.error(`version parameter must be defined.`);
+                return false;
+            }
+            if (!cmcdParametersFromManifest.keys) {
+                logger.error(`keys parameter must be defined.`);
+                return false;
+            }
+        }
+        return cmcdParametersFromManifest.version ? true : settings.get().streaming.cmcd && settings.get().streaming.cmcd.enabled;
+    }
+
+    function _checkIncludeInRequests(cmcdParametersFromManifest) {
+        let enabledRequests = settings.get().streaming.cmcd.includeInRequests;
+
+        if (cmcdParametersFromManifest.version) {
+            enabledRequests = cmcdParametersFromManifest.includeInRequests ?? [CMCD_ALL_REQUESTS_PLACEHOLDER];
```

Comment:
The default value for `includeInRequests` shall be `segments`. See also the comment from @ZmGorynych